ggml/src/ggml-sycl/ggml-sycl.cpp

```diff
@@ -1795,10 +1789,15 @@ static  void pool2d_nchw_kernel(
         const int ew = sycl::min(iw, start_w + kw);
 
         To res = 0;
+        bool op_valid = true;
 
         switch (op) {
             case GGML_OP_POOL_AVG: res = 0; break;
             case GGML_OP_POOL_MAX: res = -FLT_MAX; break;
+            default:
+                res      = NAN;
```

Comment:
I don't think it is safe to use `NAN` inside a SYCL kernel. The spec conformant way is `sycl::nan(uint32_t(0))` to get a float NaN which I think you can cast to `To`. See https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#api:nan

Reply:
Nice catch, thanks!